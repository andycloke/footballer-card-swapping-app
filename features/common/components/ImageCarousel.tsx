import * as React from 'react';
import {
  StyleSheet,
  Dimensions,
  Image,
  View,
  ImageSourcePropType
} from 'react-native';
import SwiperFlatList from 'react-native-swiper-flatlist';

type Props = {
  images: Array<ImageSourcePropType>;
  initialIndex?: number;
  backgroundColor?: string;
  imageBackgroundColor?: string;
  handleIndexChange?: (index: number) => void;
};

class ImageCarousel extends React.PureComponent<Props> {
  static defaultProps = {
    initialIndex: 0,
    backgroundColor: 'white',
    imageBackgroundColor: 'white'
  };
  handleIndexChange = ({ index }: { index: number }) => {
    const { handleIndexChange } = this.props;
    if (handleIndexChange && typeof handleIndexChange === 'function') {
      handleIndexChange(index);
    }
  };
  render() {
    const {
      images,
      backgroundColor,
      imageBackgroundColor,
      initialIndex
    } = this.props;
    return (
      <View style={[styles.container, { backgroundColor }]}>
        <SwiperFlatList
          index={initialIndex}
          onMomentumScrollEnd={this.handleIndexChange}
        >
          {images.map((image: ImageSourcePropType, i: number) => (
            <View key={i} style={styles.child}>
              <Image
                source={image}
                style={[
                  styles.image,
                  {
                    backgroundColor: imageBackgroundColor
                  }
                ]}
              />
            </View>
          ))}
        </SwiperFlatList>
      </View>
    );
  }
}

export const { width, height } = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-end'
  },
  child: {
    height,
    width,
    alignItems: 'center',
    justifyContent: 'center'
  },
  image: {
    width: width,
    height: height * 0.6,
    resizeMode: 'cover'
  }
});

export default ImageCarousel;
