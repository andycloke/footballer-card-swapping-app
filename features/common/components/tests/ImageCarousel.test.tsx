import * as React from 'react';
import * as renderer from 'react-test-renderer';

import ImageCarousel from '../ImageCarousel';
import exampleImages from './data';

describe('ImageCarousel', () => {
  it('renders a list of images', () => {
    const rendered = renderer
      .create(<ImageCarousel images={exampleImages} />)
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
