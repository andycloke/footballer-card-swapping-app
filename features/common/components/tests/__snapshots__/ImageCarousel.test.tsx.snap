// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel renders a list of images 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View>
    <RCTScrollView
      autoplay={false}
      autoplayDelay={3}
      autoplayDirection={true}
      autoplayLoop={false}
      data={
        Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 800.4,
                    "resizeMode": "cover",
                    "width": 712.5,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 800.4,
                    "resizeMode": "cover",
                    "width": 712.5,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 800.4,
                    "resizeMode": "cover",
                    "width": 712.5,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
          </Component>,
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      index={0}
      initialNumToRender={1}
      invertStickyHeaders={undefined}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onScrollToIndexFailed={[Function]}
      pagingEnabled={true}
      renderAll={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 800.4,
                    "resizeMode": "cover",
                    "width": 712.5,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "width": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
